### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by AleksejC**
Ability to save the task(after entering its name and description). Then it gets in "To do" column
Снимок экрана (13)
Снимок экрана (14)
Снимок экрана (15)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
-import {ToDoList} from "./ToDoList";
+import {ToDoList} from './ToDoList/ToDoList';
+import {TaskCard} from './Task/TaskCard';
 
 function App() {
   return (
-      <ToDoList />
+      <>
```
Comment (dimitryc-ytree) [inline src/App.tsx]: revert

File: src/Task/TaskCard.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {TaskProps} from "./types";
+import './TaskCard.css';
+
+export const TaskCard: React.FC<TaskProps> = ({ task: { name, description } }) => {
+
+    return (
```
Comment (dimitryc-ytree) [inline src/Task/TaskCard.tsx]: unnecessary return (arrow function could be inlined)

File: src/ToDoList/ToDoList.tsx [hunk 3]
```diff
@@ -1,29 +1,46 @@
-import React from 'react';
-import {useCallback} from "react";
-import {useState} from "react";
+import React, {useCallback, useState} from 'react';
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]: please configure 'Code style' section in WebStorm settings to have next style (add spaces inside curly braces):
```suggestion
import React, { useCallback, useState } from 'react';
```

File: src/Task/TaskCard.css [hunk 4]
```diff
@@ -0,0 +1,24 @@
+.task-window {
+    min-height: 150px;
+    width: 400px;
+    height: 150px;
```
Comment (dimitryc-ytree) [inline src/Task/TaskCard.css]: there is no need to specify `height` if `min-height` is specified

File: src/ToDoList/ToDoList.tsx [hunk 5]
```diff
@@ -1,29 +1,46 @@
...
+
+    const clearAllInputs = () => {
+        setCurrentTaskName('');
+        setCurrentTaskDescription('');
+    }
+
     const closeAllModals = () => {
         toggleAddNewTaskModal();
         toggleCancelConfirmModal();
+        clearAllInputs()
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]:
```suggestion
        clearAllInputs();
```

File: src/ToDoList/ToDoList.tsx [hunk 6]
```diff
@@ -34,6 +51,9 @@ export const ToDoList = () => {
                 <div className='header-and-column'>
                     <h1 className='column-header'>TO DO</h1>
                     <div className='column'>
+                        {toDoTasks.map((it) => {
+                            return <TaskCard task={it} />
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]: 1. You should provide `key` prop to each item when render components via `.map()`.
2. Unnecessary return in arrow function.

File: src/ToDoList/ToDoList.tsx [hunk 7]
```diff
@@ -68,28 +88,29 @@ export const ToDoList = () => {
                                 <label htmlFor='standard-helper-text'>New task name</label>
                             </div>
                             <TextField
+                                onChange={event => setCurrentTaskName(event.currentTarget.value)}
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]: useCallback

File: src/ToDoList/ToDoList.tsx [hunk 8]
```diff
@@ -68,28 +88,29 @@ export const ToDoList = () => {
...
-                                helperText="0/100 characters"
+                                helperText={`${currentTaskName.length}/100 characters`}
                                 variant="standard"
                                 size='medium'
                             />
                             <div>
                                 <label htmlFor='outlined-multiline-static'>Description</label>
                             </div>
                             <TextField
+                                onChange={event => setCurrentTaskDescription(event.currentTarget.value)}
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]: useCallback

**Review by dimitryc-ytree — CHANGES_REQUESTED** [review]

**Review by dimitryc-ytree — CHANGES_REQUESTED** [review]
I can't see task `id`

File: src/ToDoList/ToDoList.tsx (hunk 3 above)
Comment (AleksejC) [reply src/ToDoList/ToDoList.tsx]: Which must be checked? (except import/export braces)
Снимок экрана (16)

File: src/Task/TaskCard.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+                    <div className='task-id'>
+                            {id}
+                    </div>
+                    <div className='task-name'>
+                            {name}
+                    </div>
+                    <div className='task-description'>
+                            {description}
+                    </div>
+            </div>
```
Comment (dimitryc-ytree) [inline src/Task/TaskCard.tsx]:
```suggestion
export const TaskCard: React.FC<TaskProps> = ({ task: { name, description, id} }) => (
    <div className='task-window'>
        <div className='task-id'>
            {id}
        </div>
        <div className='task-name'>
            {name}
        </div>
        <div className='task-description'>
            {description}
        </div>
    </div>
);
```

File: src/ToDoList/ToDoList.tsx [hunk 10]
```diff
@@ -34,6 +60,12 @@ export const ToDoList = () => {
                 <div className='header-and-column'>
                     <h1 className='column-header'>TO DO</h1>
                     <div className='column'>
+                        {toDoTasks.map((it) =>
+                            <TaskCard
+                                task={it}
+                                key={it.id}
+                            />
+                        )}
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]:
```suggestion
                        {
                            toDoTasks.map(it => (
                                <TaskCard
                                    task={it}
                                    key={it.id}
                                />
                            )
                        }
```

File: src/ToDoList/ToDoList.tsx [hunk 11]
```diff
@@ -1,31 +1,57 @@
...
+    }
+
     const closeAllModals = () => {
         toggleAddNewTaskModal();
         toggleCancelConfirmModal();
+        clearAllInputs();
     }
 
-    const onSaveButtonClick = () => {
+    const id = parseFloat(Math.random().toFixed(4));
```
Comment (dimitryc-ytree) [inline src/ToDoList/ToDoList.tsx]: should be like this: "T-1", "T-2", etc.

**Review by dimitryc-ytree — CHANGES_REQUESTED** [review]

File: src/ToDoList/ToDoList.tsx (hunk 3 above)
Comment (dimitryc-ytree) [reply src/ToDoList/ToDoList.tsx]: Screenshot 2023-10-19 at 23 20 01

File: src/Task/types.ts [hunk 12]
```diff
@@ -2,7 +2,7 @@ export type Task = {
     name: string;
     description: string;
     status: Status;
-    id: number;
+    id: number | string;
```
Comment (dimitryc-ytree) [inline src/Task/types.ts]:
```suggestion
    id: string;
```

**Review by dimitryc-ytree — APPROVED** [review]

**Pull request merged**